cohere/cohere.go

```diff
@@ -130,13 +135,12 @@ func (e *CohereEmbeddingFunction) CreateEmbedding(documents []string) ([][]float
 	return response.Embeddings, nil
 }
 
-func (e *CohereEmbeddingFunction) CreateEmbeddingWithModel(documents []string, model string) ([][]float32, error) {
+func (e *CohereEmbeddingFunction) EmbedQuery(ctx context.Context, document string) ([]float32, error) {
 	response, err := e.apiClient.CreateEmbedding(&CreateEmbeddingRequest{
-		Model: model,
-		Texts: documents,
+		Texts: []string{document},
```

Comment:
let's also pass the context to CreateEmbedding and then pass it to the HTTP request

Reply:
Done